config/config.go

```diff
@@ -294,56 +304,79 @@ type AgentConfig struct {
 	BufferDirectory string `toml:"buffer_directory"`
 }
 
-// InputNames returns a list of strings of the configured inputs.
-func (c *Config) InputNames() []string {
-	name := make([]string, 0, len(c.Inputs))
+// getPluginPrintString returns a string representation of the plugin names
+// based on the NewPluginPrintBehaviour flag
+func getPluginPrintString(plugins pluginNames) string {
+	output := PluginNameCounts(plugins)
+	if PrintPluginConfigSource {
+		return output + plugins.String()
+	}
+
+	return output
+}
+
+// InputNames returns a string of configured inputs.
+func (c *Config) InputNames() string {
+	plugins := make(pluginNames, 0, len(c.Inputs))
 	for _, input := range c.Inputs {
-		name = append(name, input.Config.Name)
+		plugins = append(plugins, pluginPrinter{
+			name:   input.Config.Name,
+			source: input.Config.Source,
+		})
 	}
-	return PluginNameCounts(name)
+	return getPluginPrintString(plugins)
 }
```

Comment:
How about keeping this as is and add a function `InputNamesWithSources()` and call this in `telegraf.go`? Same for the other plugin types. This simplifies the code I think and we can keep the short printing in `telegraf.go` and **add** printing the sources.

Reply:
Made the changes.